core/modules/util/ConfigStore.h

```diff
@@ -76,31 +89,60 @@ class ConfigStore {
     friend std::ostream& operator<<(std::ostream &out, ConfigStore const& config);
 
     /** Get value for a configuration key
+     *
      * @param key configuration key
      * @return the string value for a key
-     * @throw a ConfigError exception if key is not found
+     * @throw KeyNotFoundError if key is not found
      */
-    std::string get(std::string const& key) const;
+    std::string getString(std::string const& key) const;
 
-    /** Get value for a configuration key
+    /** Get value for a configuration key or a default value if key is not found
+     *
      * @param key configuration key
      * @params defaultValue
      * @return the string value for a key, defaulting to defaultValue
      */
-    std::string get(std::string const& key, std::string const& defaultValue) const;
+    std::string getStringOrDefault(std::string const& key,
+        std::string const& defaultValue = std::string()) const;
 
-    /// @return the typed value for a key, defaulting to defaultValue
-    int getInt(std::string const& key, int const& defaultValue = 0) const;
+    /** Get value for a configuration key or a default value if key is not found
+     *
+     * @param key configuration key
+     * @params defaultValue
+     * @return the integer value for a key, defaulting to defaultValue
+     *
+     * @throw InvalidIntegerValue if value can not be converted to an integer
+     */
+    int getIntOrDefault(std::string const& key, int const& defaultValue = 0) const;
```

Comment:
Why not:

    // no default, will throw if missing
    int getInt(string key);
    // user gives a default, used if missing
    int getInt(string key, int defVal);